### ISSUE TITLE
Add Classifier estimator and Categorical prior for discrete parameters

### ISSUE THREAD

**Issue by cweniger** [cweniger/falcon#102]
## Summary

Add support for discrete categorical parameters via a new `Classifier` estimator and `Categorical` prior.

## Motivation

Currently all estimators (`Flow`, `GaussianFullCov`) assume continuous parameters. Many real-world SBI problems involve discrete latent variables (e.g. model selection, class labels, integer-valued parameters).

### `falcon/priors/categorical.py` — `Categorical`

A simple prior over D independent categorical variables each with K classes.

- `K: int` — number of classes per variable
- `D: int = 1` — number of independent categorical variables
- `probs=None` — prior class probabilities; `None` = uniform; shape `[K]` shared across D variables; shape `[D, K]` per-variable
- `param_dim = D`
- `simulate_batch(n)` → integer numpy array of shape `(n, D)`, dtype `int64`, sampled according to `probs`
- Exposes `self.log_probs` of shape `(D, K)` for use by the estimator's `sample_prior`

No `forward`/`inverse` methods needed — the classifier works directly in discrete space.

### `falcon/estimators/classifier.py` — `Classifier`

A `StepwiseEstimator` that learns `p(theta | x)` for discrete theta via cross-entropy classification.

**Parameters:**
- `K: int` — number of classes
- `D: int = 1` — number of independent categorical variables
- `gamma: float = 0.5` — proposal tempering; `gamma=0` → uniform (prior), `gamma=1` → posterior
- `hidden_dim: int`, `num_layers: int` — MLP architecture per head
- `embedding` — embedding config dict, same pattern as `Flow`/`GaussianFullCov`
- Standard loop/optimizer params: `max_epochs`, `lr`, `batch_size`, `early_stop_patience`, etc.

**Architecture:** D independent MLP heads via `nn.ModuleList`, each mapping `embedding_dim → hidden → K logits`. Forward pass: `torch.stack([head(emb) for head in self.heads], dim=1)` → shape `(batch, D, K)`.

**Training:** Cross-entropy loss summed over D heads. Theta arrives as `int64` from the buffer (preserved by `torch.as_tensor`); an assertion guards against accidental float input.

**Sampling:**
- `sample_prior`: call `simulator_instance.simulate_batch`, compute `log_prob` from `simulator_instance.log_probs`
- `sample_proposal`: sample from `softmax(gamma * logits)`, return per-sample log-prob
- `sample_posterior`: correct for gamma-tempered training data via logit rescaling — `softmax(logits / gamma)` — analogous to the `(1 + gamma) / gamma` correction in `GaussianFullCov`

**Save/load:** standard `torch.save`/`torch.load` of model weights + init tensors.

## YAML Usage Example

```yaml
graph:
  label:
    evidence: [x]
    simulator:
      _target_: falcon.priors.Categorical
      K: 3
      probs: [0.2, 0.5, 0.3]   # optional; omit for uniform
    estimator:
      _target_: falcon.estimators.Classifier
      K: 3
      gamma: 0.5
      embedding:
        _target_: model.E
        _input_: [x]
```

## Files to Create/Modify

- `falcon/priors/categorical.py` (new)
- `falcon/estimators/classifier.py` (new)
- `falcon/priors/__init__.py` — export `Categorical`
- `falcon/estimators/__init__.py` — export `Classifier`
